plugins/common/kafka/config.go

```diff
@@ -102,5 +122,27 @@ func (k *Config) SetConfig(config *sarama.Config) error {
 		config.Metadata.Full = *k.MetadataFull
 	}
 
+	if k.MetadataRetryMax != 0 {
+		config.Metadata.Retry.Max = k.MetadataRetryMax
+	}
+
+	if k.MetadataRetryBackoff != 0 {
+		// If config.Metadata.Retry.BackoffFunc is set, sarama ignores
+		// config.Metadata.Retry.Backoff
+		config.Metadata.Retry.Backoff = k.MetadataRetryBackoff
+	}
+
+	switch strings.ToLower(k.MetadataRetryType) {
+	default:
+		return fmt.Errorf("invalid metadata retry type")
+	case "exponential":
+		if k.MetadataRetryBackoff != 0 {
+			k.MetadataRetryBackoff = 250 * time.Millisecond
+			log.Warnf("metadata_retry_backoff is 0, using %s", k.MetadataRetryBackoff)
+		}
```

Comment:
Hmmm, isn't the condition exactly the inverse of your warning? Like if `k.MetadataRetryBackoff` is zero the warning is not issued?

Reply:
Oops, I definitely didn't want the `!`. I guess I should add a unit test :smile:

Reply:
Fixed this, thanks for the review!